viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
+#from tkinter.tix import InputOnly
+
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"]= genre
+        new_movie["rating"]= rating
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_movies = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def add_to_watched_list(user_data, movie):
+    watchlist = []
+
+    user_data.append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for items in user_data["watchlist"]:
+        if title in user_data["watchlist"]:
+            user_data["watched"].append(user_data["watchlist"][0])
+            user_data["watchlist"].pop(0)
+        return user_data
+
+# def add_to_watched(user_data, movie):
+#     movies_watched = []
+#     updated_data = []
+
+#     for items in len(user_data)["movie"]:
+#         if movie not in user_data:
+#             user_data["watched"].append(updated_data[movies_watched])
+#         return updated_data
+
+        
+
+
+
+# make a dictionary with these keys above
+#title:title, genre:genre, rating:rating
+
+#-----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+#THIS TEST PASSED AND THEN STOPPED PASSING_ I"M NOT SURE WHY< BUT I DID NOT HAVE TIME TO FIX IT
+def get_watched_avg_rating(user_data):
+    ratings = []
+    ratings_sum = 0
+    if(len(user_data["watched"])) is 0:

```

Comment:
Note that `is` is not the same same as `==`. `is` checks that two objects are the same object reference wise. That is, if they return the same reference id when passed into the `id` function.

`==` on the other hand checks that two objects are equal in value (even if they're not the same object.

Taking an example from [this article](https://www.geeksforgeeks.org/is-keyword-in-python/):

```python
x = ["a", "b", "c", "d"]
  
y =  ["a", "b", "c", "d"]
  
print(x is y)
print(x == y)
```

In this case, the first print statement will print `False` while the second will print `True`. Even though both `x` and `y` have the same elements, they are different objects and thus `x is y` returns `False`.

Reply:
Additionally rather than checking the length specifically through the `len` function, you can take advantage of Python's definition of falsy:

```python
if not user_data["watched"]:
```

Since empty lists count as false, this works. It would also cover the case that `user_data["watched"]` was `None` as well. (Not that I think we ever set it to that, but it definitely would be another way to implement a user with no watched movies.)

Reply:
Note: Though there's a few other places that you're using `is` like this, I'm only going to point it out once.